src/main/java/com/github/tommyettinger/digital/ArrayTools.java

```diff
@@ -139,6 +139,35 @@ public static char[] charSpan(char start, char end) {
         }
         return r;
     }
+    
+    /**
+     * Stupidly simple convenience method that produces a char range from u0000 to uFFFF, limited to {@code buffer.length - 1}.
+     *
+     * @param buffer a char array that will be modified in-place; if null this returns null
+     * @return the range of chars as a char array
+     */
+    public static char[] charSpan(char[] buffer) {
+        int end;
+        if (buffer == null || (end = Math.min(buffer.length - 1, 0xFFFF)) < 0) return buffer;
+        for (char i = 0; i <= end; i++) {
```

Comment:
There's a potential infinite-loop bug here if buffer.length is 0x10000 or greater -- the for-loop variable is a `char`, so it will always be between 0 and 0xFFFF inclusive, overflowing from 0xFFFF to 0 when it should end. I'll merge this and fix, it should be quick.